src/watch-run.ts

```diff
@@ -27,7 +37,7 @@ export function makeWatchRun(instance: TSInstance) {
         }
 
         lastTimes.set(filePath, date);
-        updateFile(instance, filePath);
+        promises.push(updateFile(instance, loader, loaderIndex, filePath));
```

Comment:
What is the significance of making this array of `Promise`s vs just calling `updateFile` there and then?

Reply:
`runLoaders` called in `updateFile` is an async function. We must wait for all of them to finish before calling `callback`.
To implement waiting for multiple async functions, I think `Promise.all` is a good way.